src/app/client/src/app/modules/core/components/main-header/main-header.component.ts

```diff
@@ -188,19 +189,55 @@ export class MainHeaderComponent implements OnInit, OnDestroy {
     this.subscription = this.utilService.currentRole.subscribe(async (result) => {
       if (result) {
         this.userType = result;
+        if(this.userType!='adminstrator'){
+          this.setUserPreferences();
+          }
       } else {
         if (this.userService.loggedIn) {
           this.userService.userData$.subscribe((profileData: IUserData) => {
             if (_.get(profileData, 'userProfile.profileUserType.type')) {
               this.userType = profileData.userProfile['profileUserType']['type'];
+              if(this.userType!='adminstrator'){
+                this.setUserPreferences();
+                }
             }
           });
         }
       }
     });
+  }
 
+  getFormConfigs() {
+    this.observationUtilService.browseByCategoryForm()
+      .then((data: any) => {
+        let currentBoard = this.userPreference.framework.board[0].toLowerCase();
+        let currentUserType=this.userType.toLowerCase();
+        if (data && data[currentBoard] &&
+          data[currentBoard][currentUserType]) {
+            this.showReportMenu=true;
+        }
+        else{
+          this.showReportMenu=false;
+        }
+      });
   }
 
+  setUserPreferences() {
```

Comment:
can you put these reusable things in service and use them in both home and main-header